app/src/common/shared/org/mozilla/vrbrowser/History.java

```diff
@@ -0,0 +1,210 @@
+        JSONArray array = new JSONArray();
+        for (Item item: mItems.values()) {
+            array.put(item.toJSON());
+        }
+
+        try (FileWriter file = new FileWriter(mContext.getFilesDir().getAbsolutePath() + "/" + FILENAME)) {
+            file.write(getJSONString());
+        }
+        catch (Exception ex) {
+            Log.e(LOGTAG, "Error saving history", ex);
+        }
+    }
+
+    private void restore() {
+        try (BufferedReader reader = new BufferedReader(new FileReader(mContext.getFilesDir().getAbsolutePath() + "/" + FILENAME))) {
+            StringBuilder sb = new StringBuilder();
+            String line = reader.readLine();
+
+            while (line != null) {
+                sb.append(line);
+                sb.append("\n");
+                line = reader.readLine();
+            }
+            try {
+                JSONArray array = new JSONArray(sb.toString());
+                mItems.clear();
+                for (int i = 0; i < array.length(); ++i) {
+                    Item item = new Item(array.getJSONObject(i));
+                    if (item.url != null && item.url.length() > 0) {
+                        mItems.put(key(item.url), item);
+                    }
+                }
+
+            } catch (JSONException e) {
+                Log.e(LOGTAG, "Error parsing history", e);
+            }
+        }
+        catch (Exception ex) {
+            Log.e(LOGTAG, "Error reading history", ex);
+        }
+    }
+
+    private String getJSONString() {
+        JSONArray array = new JSONArray();
+        for (Item item: mItems.values()) {
+            array.put(item.toJSON());
+        }
+        return array.toString();
+    }
+
+    private String key(String aUri) {
+        String result = aUri.toLowerCase();
+        int index = result.indexOf('#');
+        if (index >= 0) {
+            result = result.substring(0, index);
+        }
+        return result;
+    }
+
+    private void checkMaxSize() {
+        if (mItems.size() <= MAX_ITEMS) {
+            return;
+        }
+
+        ArrayList<Item> list = new ArrayList<>(mItems.values());
+        list.sort(new Comparator<Item>() {
+            @Override
+            public int compare(Item o1, Item o2) {
+                return (int)(o1.timestamp - o2.timestamp);
+            }
+        });
+
+        for (Item item: list) {
+            mItems.remove(key(item.url));
+            if (mItems.size() >= MAX_ITEMS) {
+                return;
+            }
+        }
+    }
+
+    // GeckoSession.ProgressDelegate
+
+    @Override
+    public void onPageStart(GeckoSession geckoSession, String s) {
+
+    }
+
+    @Override
+    public void onPageStop(GeckoSession geckoSession, boolean b) {
+        String uri = SessionStore.get().getUriFromSession(geckoSession);
+        if (uri == null || uri.length() == 0) {
+            return;
+        }
+        if (uri.toLowerCase().startsWith(SessionStore.HOME_WITHOUT_REGION_ORIGIN)) {
```

Comment:
I made a [helper method](https://github.com/MozillaReality/FirefoxReality/blob/5b8a3340/app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java#L457-L459) for this purpose. you can replace this with `if (SessionStore.get().isHomeUri(uri)) {`